### PR TITLE
Limits memory requests to deploy more components

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this?**

/kind flake

During the execution of the tests, you can check that all pod started by the tests have set a **Memory Request**, and that the system memory allocated is not exhausted by examining the output of `kubectl describe node`:

```
$ kubectl describe node <worker-node>
[...]
  Namespace                                   Name                                                             CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                                   ----                                                             ------------  ----------  ---------------  -------------  ---
  cmd-link-test99ggn                          cmp-hwbaqb-app-fdc5f8b95-vfc8t                                   0 (0%)        0 (0%)      300M (1%)        300M (1%)      3s
  cmd-link-test99ggn                          service-xrusig-0                                                 201m (3%)     201m (3%)   256Mi (1%)       256Mi (1%)     3s
  cmd-service-test192vok                      nodejs-x072280828-goxk-app-6758469dd5-std6r                      0 (0%)        0 (0%)      300M (1%)        300M (1%)      4s
  cmd-service-test324vxr                      nodejs-x251149076-zthe-app-68bb4575f6-89p7v                      0 (0%)        0 (0%)      300M (1%)        300M (1%)      4s
  cmd-service-test324vxr                      redis-0                                                          201m (3%)     201m (3%)   256Mi (1%)       256Mi (1%)     4s
  cmd-service-test443cdb                      lvbphg-0                                                         201m (3%)     201m (3%)   256Mi (1%)       256Mi (1%)     3s
  cmd-service-test443cdb                      nodejs-x468076898-bsok-app-db46c7dbd-xrnsq                       0 (0%)        0 (0%)      300M (1%)        300M (1%)      4s
  cmd-service-test577rtp                      xmyqh-app-cd6fdf694-5944b                                        0 (0%)        0 (0%)      1Gi (6%)         1Gi (6%)       4s
  odo-operator-test                           gcrmypbkoh-85f9fc5b4-4nspw                                       30m (0%)      60m (1%)    128Mi (0%)       512Mi (3%)     3s
  odo-operator-test                           nodejs-x144568958-xfqz-axjsr-65f797bf5d-qdfrj                    0 (0%)        0 (0%)      300M (1%)        300M (1%)      3s
  odo-operator-test                           nodejs-x272880799-puid-cryeu-5f585f4bcf-bb6t9                    0 (0%)        0 (0%)      300M (1%)        300M (1%)      5s
  odo-operator-test                           postgresql-operator-7df4c49c98-m7cvk                             30m (0%)      60m (1%)    64Mi (0%)        128Mi (0%)     46h
  odo-operator-test                           yiglippsoo-7fbb65dbf6-sztcm                                      30m (0%)      60m (1%)    128Mi (0%)       512Mi (3%)     5s
[...]
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests           Limits
  --------           --------           ------
  cpu                2768m (46%)        883m (14%)
  memory             11219358208 (67%)  6065607168 (36%)
  ephemeral-storage  0 (0%)             0 (0%)
  hugepages-1Gi      0 (0%)             0 (0%)
  hugepages-2Mi      0 (0%)             0 (0%)
```

In this case, 67% of the node memory has been reserved.

### CODE REVIEW

File: tests/helper/helper_oc.go
Comment (dharmit) [inline tests/helper/helper_oc.go:673]: It would help if the comment and function name gave some clue about this function trying to match a regex passed by the caller.

**Review by dharmit — CHANGES_REQUESTED** [review]
@feloy All good except one request.

File: tests/helper/helper_oc.go
Comment (mohammedzee1000) [reply tests/helper/helper_oc.go:673]: Also, should we not have this in the CmdRunner interface and have kubectl implement this. We do want to get away from using specific runners right?

Looks good to me otherwise. Can lgtm once both these concerns are addressed

**Review by mohammedzee1000 — CHANGES_REQUESTED** [review]

File: tests/helper/helper_oc.go
Comment (feloy) [reply tests/helper/helper_oc.go]: @mohammedzee1000 @dharmit I have made the changes

**Review by dharmit — COMMENTED** [review]
/approve

@feloy thanks for making the changes. 👍🏾

**Comment by feloy** [comment]
/gold

I prefer to wait for the pr #4754 to be merged

File: tests/integration/operatorhub/cmd_service_test.go
Comment (prietyc123) [inline tests/integration/operatorhub/cmd_service_test.go:69]: Not sure if this will have any role here. IIRC I had discussion with @dharmit around this and `operatorhub` tests are designed for testing functionality related to operators. We are not worried about the source code of the component. @dharmit Correct me if I am wrong.

Comment (feloy) [reply tests/integration/operatorhub/cmd_service_test.go:69]: I would prefer to have these files that can help for the visibility of the logs: when running `odo push` withtout any source files, you get errors on the container logs because the build of the sources fails.

Comment (dharmit) [reply tests/integration/operatorhub/cmd_service_test.go:69]: @prietyc123 the reason given by @feloy makes sense.

**Review by dharmit — APPROVED** [review]

**Pull request merged**
